pkg/kube/inject/webhook.go

```diff
@@ -575,6 +574,25 @@ func reapplyOverwrittenContainers(finalPod *corev1.Pod, originalPod *corev1.Pod,
 	return finalPod, nil
 }
 
+func resetFieldsInAutoImageContainer(original *corev1.Container, template *corev1.Container) {
+	if original.Image != AutoImage {
+		return
+	}
+
+	original.Image = ""
+
+	// If the original pod comes with SecurityContext.RunAsUser and the template defines a value different than the default (1337),
+	// then ignore the original value and stick with the final (merged one)
+	// This is likely a scenario in OpenShift when the istio-proxy container with image: auto is parsed, if SecurityContext.RunAsUser
+	// does not exist, OpenShift automatically assigns a value which is based on an annotation in the namespace. Regardless if the user
+	// provided that value or if it was assigned by OpenShift, the correct value is the one in the template, as set by the `.ProxyUID` field.
+	if original.SecurityContext != nil && template.SecurityContext != nil && template.SecurityContext.RunAsUser != nil &&
```

Comment:
`image: auto` and `securityContext` are unrelated. A user can override the proxy container AND set `image: some-explicit-proxy-image`

Reply:
that's right.

Reply:
@howardjohn fixed.

Reply:
LGTM now although the function name is maybeconfusing